### PR TITLE
More ways for ethereals to charge

### PR DESCRIPTION
**Pull request by wesoda25**
## About The Pull Request

Ethereals can now charge through APCs and powercells! Use harm intent with an empty hand as an ethereal on an APC, and you will steal some charge from it (Note: you can't drain an APC past 50%). If you use grab intent on an APC, you will transfer your own charge into it. Simply click a powercell when its in your hand and you'll drain some power from it. To keep this balanced, you waste way more energy from the powercell than you actually receive. 

You also now get less energy from light sources. 

## Why It's Good For The Game

More ways to charge is good! Makes playing Ethereal more fun. 

## Changelog
:cl:
add: You can now charge through APC's and Energy Cells as an Ethereal! Click an APC on harm intent to drain its energy, or on grab intent to give it your own energy! Click the energy cell while its in your hand to siphon its energy. 
balance: You now receive less charge from light as an Ethereal. 
/:cl:

@Qustinnus oked this.

Later I wanna add some more to ethereals while also making these continuous do afters (or rather he wants to - but I'll probably get around to it sooner).

### CODE REVIEW

**Comment by wesoda25** [comment]
How do I stop someone from queuing a ton of do_afters? Light draining code had that issue and I just copied that tbh

**Comment by Krysonism** [comment]
> 
> 
> How do I stop someone from queuing a ton of do_afters? Light draining code had that issue and I just copied that tbh

My sutures have the same issue, people sew up 6 of their own limbs at the same time.

**Comment by Shadowflame909** [comment]
dont your sutures automatically continue without anyone pressing a button though

**Comment by wesoda25** [comment]
> dont your sutures automatically continue without anyone pressing a button though

No comment on sutures but thats what we want Ethereal draining to do

**Comment by Shadowflame909** [comment]
@wesoda25 ninja gloves literally do that

just steal that code!1!

ur ethereal idea already exists in thy space ninja

**Comment by Iatots** [comment]
Feels like the limit is making the assumption you are the only thing that drains apcs. An ethereal on a lowpower station starving to death because "oopsie woopsie can't have you do that now" doesn't sound good.

**Comment by GuyonBroadway** [comment]
Ethereal science powergamers chugging thier boomer juice to keep the sci APC running despite all the other power being out, it's just like the good old days when sci could print fully charged batteries and use them to break the laws of thermodynamics by infinitely replacing the one in the APC.

**Comment by HomocidlePancakes** [comment]
Why allow them only drain it up to 50%? If the ethereal is being a shitter, then the crew should kick his ass.

**Comment by Anonmare** [comment]
Mush mush power slave

Also I feel like emagged APCs should have a special interaction

**Comment by wesoda25** [comment]
> Feels like the limit is making the assumption you are the only thing that drains apcs. An ethereal on a lowpower station starving to death because "oopsie woopsie can't have you do that now" doesn't sound good.

APCs are not the only way for an ethereal to get power. But yes, I do like the idea that a low power station results in ethereals not having a fun time. 



> Ethereal science powergamers chugging thier boomer juice to keep the sci APC running despite all the other power being out, it's just like the good old days when sci could print fully charged batteries and use them to break the laws of thermodynamics by infinitely replacing the one in the APC.

It takes a lot of time for ethereals to fill up an APC considering how little charge they give it. This would mean an ethereal would have to waste a shit ton of time filling it up, when they could literally just set up the sm in half the time. Not an issue imo.



> Why allow them only drain it up to 50%? If the ethereal is being a shitter, then the crew should kick his ass.

Maybe, but for now I will keep it at 50. If power is low on the station I want ethereals to be forced to look elsewhere for their charge.

**Comment by Iatots** [comment]
Or they could just bash the APC and suck the cell too if they are starving.

**Comment by HomocidlePancakes** [comment]
>Maybe, but for now I will keep it at 50. If power is low on the station I want ethereals to be forced to look elsewhere for their charge.

Mate, there are powercells inside the APC. Introducing a new mechanic and then putting an artificial block on it because it can cause minor griff is just silly.

**Comment by necromanceranne** [comment]
With your other change over overcharge, my concerns with an ethereal systematically draining the station of power through the APC's is probably not going to be a big issue. Though they can definitely HASTEN this.

Not that this is a bad thing, as it is definitely a cool power ethereals could have that is distinctly flavourful. But something to keep in mind.

**Comment by wesoda25** [comment]
> With your overcharge pr, my first concerns about this pr with an ethereal systematically draining the station of power through the APC's is probably not going to be a big issue. Though they can definitely HASTEN station power drain if they wanted to do so.
> 
> Not that this is a bad thing, as it is definitely a cool power ethereals could have that is distinctly flavourful. But something to keep in mind.

Not an issue imo. battery wise ethereals have hardly any power capacity compared to the stations battery cells. Didn't do any testing but I imagine an ethereal would die before they successfully drained 3 APCs or something. If said APC's are charging I doubt they could make it past one. 

Of course, this now actually opens me up to the idea of increasing ethereal charge capacity, so that they are on par with the stations batteries. I'm not certain how ethereal stomachs work, but I would love to see people harvesting ethereal stomach batteries for their super capacity or something. But as of now, there is nothing to worry about.

File: code/modules/power/apc.dm
Comment (ATH1909) [inline code/modules/power/apc.dm]: I kind of feel like this should be a bit faster, so that maybe more people could use it?

Or just don't "fix" the "they can stack up a bunch of do_afters" "problem", and make it a feature instead.

File: code/modules/power/apc.dm
Comment (ATH1909) [inline code/modules/power/apc.dm]: do you think that an antag ethereal character would care about draining all of the power out of an APC? I feel like there should be a better explanation for this than "your character (who just put down a power sink) doesn't WANT to drain this APC dry because reasons" (maybe "your character physically CAN'T drain this APC dry"? or just remove this limitation, it's kind of arbitrary).

File: code/modules/power/cell.dm
Comment (ATH1909) [inline code/modules/power/cell.dm]: this exchange rate is a bit TOO awful, imo.

maybe make it, like, 3 gained (by you):30 lost (by the cell)?

also, what happens if the cell has less than 100 charge but more than 25?

File: code/modules/power/cell.dm
Comment (ATH1909) [inline code/modules/power/cell.dm]: maybe add a way to channel charge from you into the cell?

Comment (wesoda25) [reply code/modules/power/cell.dm]: for some reason the idea of that sounds bad to me, like somewhere down the line people will bitch about the balance aspect of it, also im lazy so like maybe idk

File: code/modules/power/cell.dm
Comment (wesoda25) [reply code/modules/power/cell.dm]: that 25 is an old value, I’ll need to change it. 

also the gain vs lost ratio is fine to me. If anything ethereals still get too much

File: code/modules/power/apc.dm
Comment (wesoda25) [reply code/modules/power/apc.dm]: speed is fine imo. ill try and fix that problem at a later date

File: code/modules/power/apc.dm
Comment (wesoda25) [reply code/modules/power/apc.dm]: i feel as if its fine, but I _could_ change the explanation to something more apc function wise, like: “the APC’s safeties prevent you from draining any power!” Or something

File: code/modules/power/apc.dm
Comment (kriskog) [inline code/modules/power/apc.dm]: Put this species check with the other one further up

File: code/modules/power/cell.dm
Comment (kriskog) [inline code/modules/power/cell.dm]: same as above, use isethereal(user) then cast to human

**Review by kriskog — CHANGES_REQUESTED** [review]
Fix these, rest seems fine.

File: code/modules/power/cell.dm
Comment (wesoda25) [reply code/modules/power/cell.dm]: still need to fix the 25 here

**Comment by wesoda25** [comment]
Fixed the 25 and did all skogs stuff, also had to update some values since max ethereal charge is higher now

**Review by kriskog — APPROVED** [review]
As with the other one, opening this should have been approved by a maintainer beforehand. You know the drill now.

**Pull request merged**
